VogtObserver2/Program.cs

```diff
@@ -1,12 +1,39 @@
 ﻿using System;
+using System.Collections.Generic;
+using System.Threading;
 
 namespace VogtObserver2
 {
     class Program
     {
         static void Main(string[] args)
         {
-            Console.WriteLine("Hello World!");
+            
+
+            Run();
+        }
+
+        public static void Run()
+        {
+            StockGrabber stockGrabber = new StockGrabber();
+
+            Trader t1 = new Trader(stockGrabber, "Julie","MSFT");
+            Trader t2 = new Trader(stockGrabber, "Amy","APPL");
+            Trader t3 = new Trader(stockGrabber, "Mark","GOOG");
+
+            stockGrabber.AddObserver(t1);
+            stockGrabber.AddObserver(t2);
+            stockGrabber.AddObserver(t3);
+
+
+            for (int i = 0; i <= 5; i++)
+            {
+
+                stockGrabber.Notify();
```

Comment:
You shouldn't be calling notify here. Thats supposed to be abstracted/encapsulated within the observable. Instead, you should be calling the stockGrabber to Fetch the latest stock prices. Within the stock grabber's function, when it grabs the latetst stock price it should push those to the observers (INTEERNALLY within tthe class! Not here)

Reply:
```c#
for (int i = 0; i <= 5; i++) {
  stockGrabber.FetchStockPrice(GetRandomStockName));
}
```

Then within that FetchStockPrice, call `Update(price, symbol)`.